programs/mpl-core/src/plugins/update_delegate.rs

```diff
@@ -102,17 +97,48 @@ impl PluginValidation for UpdateDelegate {
         }
     }
 
+    // Validate the approve plugin authority lifecycle action.
+    fn validate_approve_plugin_authority(
+        &self,
+        ctx: &PluginValidationContext,
+    ) -> Result<ValidationResult, ProgramError> {
+        let plugin = ctx.target_plugin.ok_or(MplCoreError::InvalidPlugin)?;
+
+        // If the plugin authority is the authority signing.
+        if (ctx.self_authority
+            == &(Authority::Address {
+                address: *ctx.authority_info.key,
+            })
+            // Or the authority is one of the additional delegates.
+            || self.additional_delegates.contains(ctx.authority_info.key))
+            // And it's an authority-managed plugin.
+            && plugin.manager() == Authority::UpdateAuthority
+            // And the plugin is not an UpdateDelegate plugin, because we cannot change the authority of the UpdateDelegate plugin.
+            && PluginType::from(plugin) != PluginType::UpdateDelegate
```

Comment:
In the PR description it calls out additional delegate cannot:
> change the plugin authority of the updateDelegate plugin (the root asset updateDelegate authority)

I think the requirement is covered by the `&&` condition at 116 and 117, because if the UpdateAuthority is doing it, it would be approved by the Asset or Collection version of `validate_approve_plugin_authority`.

There should probably be a test for this condition if there's not one already.

Reply:
I think that's covered by 'it cannot approve the update plugin authority as additional delegate' and 'it cannot revoke the update plugin authority as additional delegate'